core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java

```diff
@@ -481,10 +539,14 @@ public void write(String s, byte[] bytes) {
 
         MethodDescriptor registerSerializationMethod = null;
 
+        MethodCreator registerForReflection = file
+                .getMethodCreator("registerForReflection", void.class, BeforeAnalysisAccess.class)
+                .setModifiers(Modifier.PRIVATE | Modifier.STATIC);
+
         for (Map.Entry<String, ReflectionInfo> entry : reflectiveClasses.entrySet()) {
             MethodCreator mv = file.getMethodCreator("registerClass" + count++, void.class, Feature.BeforeAnalysisAccess.class);
```

Comment:
nitpick: 
```suggestion
            MethodCreator mv = file.getMethodCreator("registerClass" + count++, void.class, BeforeAnalysisAccess.class);
```

Reply:
We use `Feature.BeforeAnalysisAccess` all around so I wanted to be consistent...

Reply:
You are right. So it's the other way around, please see my new suggestions.